src/Core/src/MauiContext.cs

```diff
@@ -72,6 +72,20 @@ public void AddSpecific(Type type, Func<object, object?> getter, object state)
 			{
 				_scopeStatic[type] = (state, getter);
 			}
+
+			public object? GetKeyedService(Type serviceType, object? serviceKey)
+			{
+				if (Inner is IKeyedServiceProvider provider)
+					return provider.GetKeyedService(serviceType, serviceKey);
+
+				// we know this won't work, but we need to call it to throw the right exception
+				return Inner.GetRequiredKeyedService(serviceType, serviceKey);
```

Comment:
What do you mean throw the "right" exception? Doesn't this just throw an exception like "no service for serviceKey"? You want an error like https://github.com/dotnet/aspnetcore/blob/cc9cff31eb828f5849c07afc46b08baeda42b399/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L793